lib/Mojo/SMTP/Client.pm

```diff
@@ -326,17 +326,52 @@ sub _make_cmd_steps {
 			}
 		}
 		elsif ($cmd[$i] eq 'auth') { # AUTH
-			push @steps, $self->_new_cmd(sub {
-				my $delay = shift;
-				$self->_write_cmd('AUTH PLAIN '.b64_encode(join("\0", '', $cmd[$mi]->{login}, $cmd[$mi]->{password}), ''), CMD_AUTH);
-				$self->_read_response($delay->begin, !$prepend && $mi == $#cmd);
-				$self->{expected_code} = CMD_OK;
-			}),
-			$self->{resp_checker},
-			sub {
+			my $type = lc($cmd[$mi]->{type} // 'plain');
+			my $set_auth_ok = sub {
 				my $delay = shift;
 				$self->{authorized} = 1;
 				$delay->pass;
+			};
+			if ($type eq 'plain') {
+				push @steps, $self->_new_cmd(sub {
+					my $delay = shift;
+					$self->_write_cmd('AUTH PLAIN '.b64_encode(join("\0", '', $cmd[$mi]->{login}, $cmd[$mi]->{password}), ''), CMD_AUTH);
+					$self->_read_response($delay->begin, !$prepend && $mi == $#cmd);
+					$self->{expected_code} = CMD_OK;
+				}),
+				$self->{resp_checker},
+				$set_auth_ok;
+			}
+			elsif ($type eq 'login') {
+				push @steps,
+				# start auth
+				$self->_new_cmd(sub {
+					my $delay = shift;
+					$self->_write_cmd('AUTH LOGIN', CMD_AUTH);
+					$self->_read_response($delay->begin, 0);
+					$self->{expected_code} = CMD_MORE;
+				}),
+				$self->{resp_checker},
+				# send username
+				$self->_new_cmd(sub {
+					my $delay = shift;
+					$self->_write_cmd(''.b64_encode($cmd[$mi]->{login}, ''), CMD_AUTH);
```

Comment:
why you need this empty strings here?

Reply:
Left-over from an ealier faulty use of b64_encode, sorry.